src/libUtils/ShardSizeCalculator.cpp

```diff
@@ -55,12 +55,70 @@ uint32_t ShardSizeCalculator::CalculateShardSize(const uint32_t numberOfNodes) {
   return result[index];
 }
 
+static void GenerateShardCountsCore(const vector<uint32_t>& shardSizeValues,
+                                    uint32_t numNodesForSharding,
+                                    vector<uint32_t>& currentResult,
+                                    vector<uint32_t>& bestResult,
+                                    uint32_t& leastWaste) {
+  // If number of remaining unsharded nodes is less than minimum threshold
+  if (numNodesForSharding < shardSizeValues[0]) {
+    // Distribute these nodes among the existing shards
+    if ((numNodesForSharding > 0) && (currentResult.size() > 0)) {
+      uint32_t toAddPerShard = numNodesForSharding / currentResult.size();
+      if ((numNodesForSharding % currentResult.size()) > 0) {
+        toAddPerShard++;
+      }
+      for (auto& shardInCurrentResult : currentResult) {
+        // Don't add more nodes than the max threshold
+        uint32_t actualToAdd =
+            min(toAddPerShard, shardSizeValues[shardSizeValues.size() - 1] -
```

Comment:
trying to understand the logic here. Why is it `shardSizeValues[shardSizeValues.size() - 1]` and not the current index of the `shardSizeValues`

Reply:
I think its high limit..  we can hard code it to `2` anyways for readability.